python/cudf/cudf/_lib/nvtx/_lib/lib.pyx

```diff
@@ -0,0 +1,64 @@
+from nvtx._lib.lib cimport *
+
+
+def initialize():
+    nvtxInitialize(NULL)
+
+
+cdef class EventAttributes:
+    cdef dict __dict__
+    cdef nvtxEventAttributes_t c_obj
+
+    def __init__(self, message=None, color=None):
+        self._message = message.encode("ascii")
+        self._color = color
+        self.c_obj = nvtxEventAttributes_t(0)
+        self.c_obj.version = NVTX_VERSION
+        self.c_obj.size = NVTX_EVENT_ATTRIB_STRUCT_SIZE
+        self.c_obj.colorType = NVTX_COLOR_ARGB
+        self.c_obj.color = color
+        self.c_obj.messageType = NVTX_MESSAGE_TYPE_ASCII
+        self.c_obj.message.ascii = <const char*> self._message
+
+    @property
+    def message(self):
+        return self._message
+
+
+cdef class DomainHandle:
+    cdef dict __dict__
+    cdef nvtxDomainHandle_t c_obj
+
+    def __init__(self, name=None):
+        if name is not None:
+            self._name = name.encode("ascii")
+            self.c_obj = nvtxDomainCreateA(self._name)
+        else:
+            self._name = None
+            self.c_obj = NULL
+
+    def __dealloc__(self):
+        nvtxDomainDestroy(self.c_obj)
```

Comment:
I'm always weary of relying on `__dealloc__` here as opposed to a smart pointer especially because `self` is a Python object and `__dealloc__` runs after the interpreter shuts down from my experience.